### PR TITLE
fix: Add missing manpages to meson.build and more

### LINKED ISSUE #11862: Installing Nix 2.25.0 from the flake is missing the LaunchDaemon plist, installs test programs that fail, and other peculiarities
**Describe the bug**

```
$ nix build github:NixOS/nix/2.25.0
```

is a bit peculiar in that:

1. The build output is missing `Library/LaunchDaemons/org.nixos.nix-daemon.plist` from the result
2. The default package no longer has a `version` attribute
3. The default package is now 4x larger than it used to be (~110MB, ~400MB)
4. The `bin` directory now has `-test` programs, and unfortunately some of those fail:

```
/nix/store/9pf0gwibm5v7fy4gn8yvp730fzwc44lr-nix-fetchers-tests-2.25.0/bin/nix-fetchers-tests
Running main() from /tmp/nix-build-gtest-1.14.0.drv-0/source/googletest/src/gtest_main.cc
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from PublicKeyTest
[ RUN      ] PublicKeyTest.PublicKey_simple_from_json
unknown file: Failure
C++ exception with description "bad_optional_access" thrown in the test fixture's constructor.

[  FAILED  ] PublicKeyTest.PublicKey_simple_from_json (0 ms)
[ RUN      ] PublicKeyTest.PublicKey_simple_to_json
unknown file: Failure
C++ exception with description "bad_optional_access" thrown in the test fixture's constructor.

[  FAILED  ] PublicKeyTest.PublicKey_simple_to_json (0 ms)
[ RUN      ] PublicKeyTest.PublicKey_defaultType_from_json
unknown file: Failure
C++ exception with description "bad_optional_access" thrown in the test fixture's constructor.

[  FAILED  ] PublicKeyTest.PublicKey_defaultType_from_json (0 ms)
[ RUN      ] PublicKeyTest.PublicKey_defaultType_to_json
unknown file: Failure
C++ exception with description "bad_optional_access" thrown in the test fixture's constructor.

[  FAILED  ] PublicKeyTest.PublicKey_defaultType_to_json (0 ms)
[ RUN      ] PublicKeyTest.PublicKey_noRoundTrip_from_json
unknown file: Failure
C++ exception with description "bad_optional_access" thrown in the test fixture's constructor.

[  FAILED  ] PublicKeyTest.PublicKey_noRoundTrip_from_json (0 ms)
[----------] 5 tests from PublicKeyTest (0 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 5 tests, listed below:
[  FAILED  ] PublicKeyTest.PublicKey_simple_from_json
[  FAILED  ] PublicKeyTest.PublicKey_simple_to_json
[  FAILED  ] PublicKeyTest.PublicKey_defaultType_from_json
[  FAILED  ] PublicKeyTest.PublicKey_defaultType_to_json
[  FAILED  ] PublicKeyTest.PublicKey_noRoundTrip_from_json

 5 FAILED TESTS
```

**Expected behavior**

I expected the default package to remain suitable for installing by default in a variety of environments.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

### PR DESCRIPTION
**Pull request by roberth**
## Motivation

- Closes https://github.com/NixOS/nix/issues/11880
- Closes #11862

## Context

---

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: doc/manual/meson.build [hunk 1]
```diff
@@ -260,7 +263,7 @@ nix3_manpages = [
   'nix3-repl',
   'nix3-run',
   'nix3-search',
-  #'nix3-shell',
```
Comment (roberth) [inline doc/manual/meson.build:263]: No man pages for aliases; a meson independent problem.

- Tracked in https://github.com/NixOS/nix/issues/11430

**Comment by Mic92** [comment]
From installer-test: `  install: /nix/var/nix/profiles/default/Library/LaunchDaemons/org.nixos.nix-daemon.plist: No such file or directory`
Is this the other problem the other pull request was talking about?

**Comment by Mic92** [comment]
Ok. It's just at the moment the CI is green and now it's not.

**Comment by roberth** [comment]
It now fixes the whole issue, and the other one, as linked in the description.

**Pull request merged**